### PR TITLE
Add support for PHP 8.0

### PR DESCRIPTION
**Pull request by wells**
Adds remi-php80 and such.

### CODE REVIEW

**Comment by geerlingguy** [comment]
Hmm... the test is failing on:

```
Package 'php8.0-json' has no installation candidate
```

Looks like: https://github.com/oerdnj/deb.sury.org/issues/1439#issuecomment-704974475

> JSON is now always included, so no `php8.0-json`, it's always there

So we'll need to make sure if `php_version == '8.0'` (or better yet, if it's major version is 8 or more), the json package is removed by default...

**Comment by geerlingguy** [comment]
(Big +1 to getting this added though... we need PHP 8 support :D

**Comment by geerlingguy** [comment]
Haha, well... you pushed that commit while I was typing.

**Comment by wells** [comment]
Hey Jeff. Yes sir. Just added a set_fact for Debian builds. Fingers crossed. 🥳

File: tasks/main.yml [hunk 1]
```diff
@@ -5,6 +5,10 @@
     - "{{ role_path }}/vars/{{ ansible_os_family }}.yml"
     - "{{ role_path }}/vars/{{ ansible_os_family }}-php{{ php_version }}.yml"
 
+- name: Remove missing JSON extension for PHP 8.0 (included by default)
+  set_fact: "{{ __php_packages }}={{ __php_packages | reject('search','php8.0-json') | list }}"
+  when: __php_packages is defined and php_version == "8.0"
```
Comment (geerlingguy) [inline tasks/main.yml:10]: Can you make this use the version compare filter? https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#comparing-versions

```
php_version is version('8.0', '>=')
```

And maybe break the when up into two lines for easier scanning:

```yaml
  when:
    - __php_packages is defined
    - php_version is version('8.0', '>=')
```

**Review by geerlingguy — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 1 above)
Comment (wells) [reply tasks/main.yml:10]: Sure. Do you have a good way to test these roles locally? Is it a matter of setting up molecule or something?

Comment (geerlingguy) [reply tasks/main.yml]: Yeah; `pip3 install molecule` then `MOLECULE_DISTRO=ubuntu1804 playbook=8.0.yml molecule test`

Note that the directory for this git checkout needs to be named 'geerlingguy.php-versions' for molecule to read it properly.

Comment (wells) [reply tasks/main.yml]: 10-4 thanks.

Comment (wells) [reply tasks/main.yml]: My local test with molecule is stuck on php 7.3. That happens running the other playbooks for 7.2 and 7.4 as well.
```
MOLECULE_DISTRO=ubuntu1804 playbook=7.2.yml molecule test
MOLECULE_DISTRO=ubuntu1804 playbook=7.4.yml molecule test
MOLECULE_DISTRO=ubuntu1804 playbook=8.0.yml molecule test
```

Comment (geerlingguy) [reply tasks/main.yml]: Oh shoot, that should actually be `MOLECULE_PLAYBOOK` not `playbook`. Sorry about that!

Comment (wells) [reply tasks/main.yml]: No worries. Thanks for the assist.

**Review by geerlingguy — APPROVED** [review]

**Comment by geerlingguy** [comment]
Thanks!

**Comment by itsmebhupendra** [comment]
I am still facing the same issue for "php8.1-json" while vagrant up.
**My configuration:**
"geerlingguy/drupal-vm": "^6.0" in composer.json which installs version "6.0.4".

**Inside config.yml:**
"vagrant_box: geerlingguy/ubuntu2004"
php_version: '8.1'

**Pull request merged**